packages/openneuro-app/src/scripts/datalad/routes/snapshot-content.jsx

```diff
@@ -60,72 +65,105 @@ const getSnapshotDetails = gql`
 `
 
 const SnapshotContent = ({ dataset, tag }) => (
-  <Query
-    query={getSnapshotDetails}
-    variables={{
-      datasetId: dataset.id,
-      tag,
-    }}>
-    {({ loading, error, data }) => {
-      if (loading) {
-        return <Spinner text="Loading Snapshot" active />
-      } else if (error) {
-        throw new Error(error)
-      } else {
-        return <SnapshotDetails dataset={dataset} snapshot={data.snapshot} />
-      }
-    }}
-  </Query>
+  <ErrorBoundary subject={'snapshot content unavailable'}>
+    <Query
+      query={getSnapshotDetails}
+      variables={{
+        datasetId: dataset.id,
+        tag,
+      }}>
+      {({ loading, error, data }) => {
+        if (loading) {
+          return <Spinner text="Loading Snapshot" active />
+        } else {
+          if(error && error.message !== 'GraphQL error: Internal Server Error')
+            throw new Error(error)
+          return <SnapshotDetails dataset={dataset} snapshot={data && data.snapshot} />
+        }
+      }}
+    </Query>
+  </ErrorBoundary>
 )
 
 SnapshotContent.propTypes = {
   dataset: PropTypes.object,
   tag: PropTypes.string,
 }
 
+const StyleContainer = styled.div({
+  '.fileupload-btn': {
+    marginTop: '2rem',
+    maxWidth: 'none',
+    lineHeight: 'normal',
+  }
+})
+
+
+const snapshotDescription = (snapshot, property, defaultValue) => ((snapshot && snapshot.description && snapshot.description[property]) || defaultValue)
+
 const SnapshotDetails = ({ dataset, snapshot }) => {
+  const snapshotName = snapshotDescription(snapshot, 'Name', 'No Snapshot')
+  
   return (
     <span>
       <div className="col-xs-6">
         <Helmet>
           <title>
-            {pageTitle} - {snapshot.description.Name}
+            {pageTitle} - {snapshotName}
           </title>
-          <meta name="description" content={snapshot.readme} />
-          <script type="application/ld+json">
-            {schemaGenerator(snapshot)}
-          </script>
+          {snapshot ? <>
```

Comment:
I'm not sure about adding conditionals here for the entire snapshot, this component is meant to always receive some kind of snapshot and while making fields conditional might be necessary - if the whole snapshot is missing we should extract the logic to a parent component rather than add conditionals to the already pretty complex template here.

Reply:
refactored